prover-service/src/witness_gen.rs

```diff
@@ -17,13 +18,24 @@ impl PathStr for NamedTempFile {
     }
 }
 
-pub fn witness_gen(config: &ProverServiceConfig, body: &str) -> Result<NamedTempFile> {
+pub fn witness_gen(
+    config: &ProverServiceConfig,
+    circuit_input_signals: &CircuitInputSignals<Padded>,
+) -> Result<NamedTempFile> {
     let _span = logging::new_span("GenerateWitness");
 
+    let formatted_input_str = serde_json::to_string(&circuit_input_signals.to_json_value())
+        .map_err(anyhow::Error::new)?;
+
+    // Only sensitive values to disk.
+    if config.enable_dangerous_logging {
```

Comment:
Can you please 🙏 explain in a comment above:
1. What is "dangerous logging"?
2. Why it may be enabled or disabled?

Reply:
not sure why the original author did this.
should it be a separate investigation?